go/mcap/writer.go

```diff
@@ -607,10 +607,21 @@ func (w *Writer) Close() error {
 			}
 		}
 	}
+	var summaryCrc uint32
+	if summarySectionStart != 0 && w.opts.IncludeCRC {
+		// add partial footer record to end of summary section CRC
+		footerPrefix := make([]byte, 1+8+8+8)
```

Comment:
there's a buffer for this, `w.msg` -- maybe should be `w.record`. You can ensure you have enough room in it with `w.ensureSized(1+8+8+8)`.